### ISSUE TITLE
Unable to activate extension : table already exist

### ISSUE THREAD

**Issue by Hiobi** [FriendsOfFlarum/linguist#10]
Hi,

When I try to (re-)enable **Linguist**, I got the following error (logs file). The activation failed.
I tried to uninstall/reinstall, the issue still here. I didn't erase old MySQL tables for the moment.

## Details

* **Flarum version** : 0.1.0-beta.8.1
* **PHP version** : 7.3.5-1
* **MySQL version**: 5.6.43-84.3
* **Extension version** : Flagrow Linguist 0.3.1

```bash
Flarum core 0.1.0-beta.8.1
PHP version: 7.3.5-1+0~20190503093827.38+stretch~1.gbp60a41b
Loaded extensions: Core, date, libxml, openssl, pcre, zlib, filter, hash, pcntl, Reflection, SPL, sodium, session, standard, mysqlnd, PDO, xml, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, imagick, json, exif, mysqli, newrelic, pdo_mysql, pdo_sqlite, Phar, posix, readline, recode, shmop, SimpleXML, sockets, sqlite3, sysvmsg, sysvsem, sysvshm, tidy, tokenizer, wddx, xmlreader, xmlrpc, xmlwriter, xsl, zip, Zend OPcache
+---------------------------------+-----------------+------------------------------------------+
| Flarum Extensions               |                 |                                          |
+---------------------------------+-----------------+------------------------------------------+
| ID                              | Version         | Commit                                   |
+---------------------------------+-----------------+------------------------------------------+
| flarum-statistics               | v0.1.0-beta.8   |                                          |
| flarum-sticky                   | v0.1.0-beta.8   |                                          |
| flarum-subscriptions            | v0.1.0-beta.8   |                                          |
| flarum-suspend                  | v0.1.0-beta.8   |                                          |
| flarum-tags                     | v0.1.0-beta.8.2 |                                          |
| flarum-flags                    | v0.1.0-beta.8.1 |                                          |
| flarum-emoji                    | v0.1.0-beta.8   |                                          |
| flarum-bbcode                   | v0.1.0-beta.8   |                                          |
| flarum-approval                 | v0.1.0-beta.8   |                                          |
| flarum-mentions                 | v0.1.0-beta.8.1 |                                          |
| flarum-markdown                 | v0.1.0-beta.8   |                                          |
| flarum-lock                     | v0.1.0-beta.8   |                                          |
| flarum-likes                    | v0.1.0-beta.8.1 |                                          |
| flarum-pusher                   | v0.1.0-beta.8.1 |                                          |
| milescellar-lang-french         | v1.2.1          |                                          |
| noriods-auto-more               | 0.3.0           |                                          |
| flagrow-direct-links            | 0.2.0           |                                          |
| flagrow-terms                   | 0.3.1           |                                          |
| flagrow-users-list              | 0.1.2           |                                          |
| reflar-polls                    | 1.3.3           |                                          |
| flagrow-upload                  | 0.7.1           |                                          |
| flagrow-byobu                   | 0.2.2           |                                          |
| fof-socialprofile               | 0.1.1           |                                          |
| fof-pages                       | 0.2.2           |                                          |
| fof-formatting                  | 0.1.2           |                                          |
| kvothe-pipetables               | 1.0.9           |                                          |
| fof-links                       | 0.1.1           |                                          |
| kvothe-keyboard-shortcuts       | v0.1.0          |                                          |
| fof-default-group               | 0.1.0           |                                          |
| flagrow-analytics               | 0.8.0           |                                          |
| jordanjay29-summaries           | 0.3.1           |                                          |
| flagrow-user-directory          | 0.2.1           |                                          |
| amaurycarrade-syndication       | v0.2.0          |                                          |
| fof-merge-discussions           | 0.2.1           |                                          |
| flagrow-split                   | 0.3.2           |                                          |
| reflar-reactions                | 1.0.0-beta.5.2  |                                          |
| fof-prevent-necrobumping        | 0.1.3           |                                          |
| michaelbelgium-discussion-views | v0.1.0-beta.8.2 |                                          |
| fajuu-contactbutton             | v2.0            |                                          |
| fof-sentry                      | 0.1.1           |                                          |
| michaelbelgium-profile-views    | v0.1.0-beta-8-4 | d4af84a8c6ea15408e6ae00942fe4f4c9fa8288b |
| fof-recaptcha                   | 0.1.0           |                                          |
| reflar-cookie-consent           | 0.2.0           |                                          |
| fof-secure-https                | v0.1.0          |                                          |
| fof-spamblock                   | 0.2.1           |                                          |
| reflar-pwned-passwords          | 0.2.0           |                                          |
| flagrow-bazaar                  | 0.3.10          |                                          |
| fof-transliterator              | 0.1.1           |                                          |
| reflar-stopforumspam            | 0.1.5           |                                          |
| flagrow-sitemap                 | 0.2.0           |                                          |
| flarum-akismet                  | v0.1.0-beta.8   |                                          |
| flagrow-impersonate             | 0.2.0           |                                          |
| rooaaar-lang-french-extended    | dev-master      | 8059489b0a8b430961a8cf5d916a8980c5acea45 |
+---------------------------------+-----------------+------------------------------------------+
Base URL: https://forum.nefald.fr
```

## Logs file
```bash
PDOException: SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'flarum_flagrow_linguist_strings' already exists
#58 vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php(117): execute
#57 vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php(117): execute
#56 vendor/illuminate/database/Connection.php(458): Illuminate\Database\{closure}
#55 vendor/illuminate/database/Connection.php(657): runQueryCallback
#54 vendor/illuminate/database/Connection.php(624): run
#53 vendor/illuminate/database/Connection.php(459): statement
#52 vendor/illuminate/database/Schema/Blueprint.php(86): build
#51 vendor/illuminate/database/Schema/Builder.php(252): build
#50 vendor/illuminate/database/Schema/Builder.php(165): create
#49 vendor/flagrow/linguist/migrations/2017_10_04_create_flagrow_linguist_strings_table.php(16): {closure}
#48 vendor/illuminate/filesystem/Filesystem.php(0): call_user_func
#47 vendor/flarum/core/src/Database/Migrator.php(197): runClosureMigration
#46 vendor/flarum/core/src/Database/Migrator.php(130): runUp
#45 vendor/flarum/core/src/Database/Migrator.php(113): runMigrationList
#44 vendor/flarum/core/src/Database/Migrator.php(87): run
#43 vendor/flarum/core/src/Extension/ExtensionManager.php(241): migrate
#42 vendor/flarum/core/src/Extension/ExtensionManager.php(127): enable
#41 vendor/flarum/core/src/Api/Controller/UpdateExtensionController.php(49): handle
#40 vendor/flarum/core/src/Http/RouteHandlerFactory.php(40): Flarum\Http\{closure}
#39 vendor/flarum/core/src/Http/Middleware/DispatchRoute.php(67): process
#38 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#37 vendor/reflar/stopforumspam/src/Middleware/RegisterMiddleware.php(75): process
#36 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#35 vendor/reflar/pwned-passwords/src/Middleware/CheckPassword.php(66): process
#34 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#33 vendor/fof/secure-https/src/Middlewares/ContentSecurityPolicyMiddleware.php(21): process
#32 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#31 vendor/fof/sentry/src/Middleware/HandleErrorsWithSentry.php(19): process
#30 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#29 vendor/flagrow/terms/src/Middlewares/RegisterMiddleware.php(59): process
#28 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#27 vendor/flarum/core/src/Http/Middleware/SetLocale.php(51): process
#26 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#25 vendor/flarum/core/src/Http/Middleware/AuthenticateWithHeader.php(55): process
#24 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#23 vendor/flarum/core/src/Http/Middleware/AuthenticateWithSession.php(34): process
#22 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#21 vendor/flarum/core/src/Http/Middleware/RememberFromCookie.php(52): process
#20 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#19 vendor/flarum/core/src/Http/Middleware/StartSession.php(62): process
#18 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#17 vendor/flarum/core/src/Api/Middleware/FakeHttpMethods.php(31): process
#16 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#15 vendor/flarum/core/src/Http/Middleware/ParseJsonBody.php(29): process
#14 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#13 vendor/flarum/core/src/Api/Middleware/HandleErrors.php(42): process
#12 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#11 vendor/zendframework/zend-stratigility/src/MiddlewarePipe.php(83): process
#10 vendor/middlewares/request-handler/src/RequestHandler.php(84): process
#9 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#8 vendor/middlewares/base-path-router/src/BasePathRouter.php(97): process
#7 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#6 vendor/middlewares/base-path/src/BasePath.php(53): process
#5 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#4 vendor/zendframework/zend-stratigility/src/MiddlewarePipe.php(83): process
#3 vendor/zendframework/zend-stratigility/src/MiddlewarePipe.php(72): handle
#2 vendor/zendframework/zend-httphandlerrunner/src/RequestHandlerRunner.php(95): run
#1 vendor/flarum/core/src/Http/Server.php(46): listen
#0 public/index.php(22): null

Doctrine\DBAL\Driver\PDOException: SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'flarum_flagrow_linguist_strings' already exists
#57 vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOStatement.php(119): execute
#56 vendor/illuminate/database/Connection.php(458): Illuminate\Database\{closure}
#55 vendor/illuminate/database/Connection.php(657): runQueryCallback
#54 vendor/illuminate/database/Connection.php(624): run
#53 vendor/illuminate/database/Connection.php(459): statement
#52 vendor/illuminate/database/Schema/Blueprint.php(86): build
#51 vendor/illuminate/database/Schema/Builder.php(252): build
#50 vendor/illuminate/database/Schema/Builder.php(165): create
#49 vendor/flagrow/linguist/migrations/2017_10_04_create_flagrow_linguist_strings_table.php(16): {closure}
#48 vendor/illuminate/filesystem/Filesystem.php(0): call_user_func
#47 vendor/flarum/core/src/Database/Migrator.php(197): runClosureMigration
#46 vendor/flarum/core/src/Database/Migrator.php(130): runUp
#45 vendor/flarum/core/src/Database/Migrator.php(113): runMigrationList
#44 vendor/flarum/core/src/Database/Migrator.php(87): run
#43 vendor/flarum/core/src/Extension/ExtensionManager.php(241): migrate
#42 vendor/flarum/core/src/Extension/ExtensionManager.php(127): enable
#41 vendor/flarum/core/src/Api/Controller/UpdateExtensionController.php(49): handle
#40 vendor/flarum/core/src/Http/RouteHandlerFactory.php(40): Flarum\Http\{closure}
#39 vendor/flarum/core/src/Http/Middleware/DispatchRoute.php(67): process
#38 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#37 vendor/reflar/stopforumspam/src/Middleware/RegisterMiddleware.php(75): process
#36 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#35 vendor/reflar/pwned-passwords/src/Middleware/CheckPassword.php(66): process
#34 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#33 vendor/fof/secure-https/src/Middlewares/ContentSecurityPolicyMiddleware.php(21): process
#32 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#31 vendor/fof/sentry/src/Middleware/HandleErrorsWithSentry.php(19): process
#30 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#29 vendor/flagrow/terms/src/Middlewares/RegisterMiddleware.php(59): process
#28 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#27 vendor/flarum/core/src/Http/Middleware/SetLocale.php(51): process
#26 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#25 vendor/flarum/core/src/Http/Middleware/AuthenticateWithHeader.php(55): process
#24 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#23 vendor/flarum/core/src/Http/Middleware/AuthenticateWithSession.php(34): process
#22 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#21 vendor/flarum/core/src/Http/Middleware/RememberFromCookie.php(52): process
#20 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#19 vendor/flarum/core/src/Http/Middleware/StartSession.php(62): process
#18 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#17 vendor/flarum/core/src/Api/Middleware/FakeHttpMethods.php(31): process
#16 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#15 vendor/flarum/core/src/Http/Middleware/ParseJsonBody.php(29): process
#14 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#13 vendor/flarum/core/src/Api/Middleware/HandleErrors.php(42): process
#12 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#11 vendor/zendframework/zend-stratigility/src/MiddlewarePipe.php(83): process
#10 vendor/middlewares/request-handler/src/RequestHandler.php(84): process
#9 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#8 vendor/middlewares/base-path-router/src/BasePathRouter.php(97): process
#7 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#6 vendor/middlewares/base-path/src/BasePath.php(53): process
#5 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#4 vendor/zendframework/zend-stratigility/src/MiddlewarePipe.php(83): process
#3 vendor/zendframework/zend-stratigility/src/MiddlewarePipe.php(72): handle
#2 vendor/zendframework/zend-httphandlerrunner/src/RequestHandlerRunner.php(95): run
#1 vendor/flarum/core/src/Http/Server.php(46): listen
#0 public/index.php(22): null

Illuminate\Database\QueryException: SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'flarum_flagrow_linguist_strings' already exists (SQL: create table `flarum_flagrow_linguist_strings` (`id` int unsigned not null auto_increment primary key, `key` varchar(180) not null, `locale` varchar(10) null, `value` text not null, `created_at` timestamp null, `updated_at` timestamp null) default character set utf8mb4 collate utf8mb4_unicode_ci engine = InnoDB)
#55 vendor/illuminate/database/Connection.php(664): runQueryCallback
#54 vendor/illuminate/database/Connection.php(624): run
#53 vendor/illuminate/database/Connection.php(459): statement
#52 vendor/illuminate/database/Schema/Blueprint.php(86): build
#51 vendor/illuminate/database/Schema/Builder.php(252): build
#50 vendor/illuminate/database/Schema/Builder.php(165): create
#49 vendor/flagrow/linguist/migrations/2017_10_04_create_flagrow_linguist_strings_table.php(16): {closure}
#48 vendor/illuminate/filesystem/Filesystem.php(0): call_user_func
#47 vendor/flarum/core/src/Database/Migrator.php(197): runClosureMigration
#46 vendor/flarum/core/src/Database/Migrator.php(130): runUp
#45 vendor/flarum/core/src/Database/Migrator.php(113): runMigrationList
#44 vendor/flarum/core/src/Database/Migrator.php(87): run
#43 vendor/flarum/core/src/Extension/ExtensionManager.php(241): migrate
#42 vendor/flarum/core/src/Extension/ExtensionManager.php(127): enable
#41 vendor/flarum/core/src/Api/Controller/UpdateExtensionController.php(49): handle
#40 vendor/flarum/core/src/Http/RouteHandlerFactory.php(40): Flarum\Http\{closure}
#39 vendor/flarum/core/src/Http/Middleware/DispatchRoute.php(67): process
#38 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#37 vendor/reflar/stopforumspam/src/Middleware/RegisterMiddleware.php(75): process
#36 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#35 vendor/reflar/pwned-passwords/src/Middleware/CheckPassword.php(66): process
#34 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#33 vendor/fof/secure-https/src/Middlewares/ContentSecurityPolicyMiddleware.php(21): process
#32 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#31 vendor/fof/sentry/src/Middleware/HandleErrorsWithSentry.php(19): process
#30 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#29 vendor/flagrow/terms/src/Middlewares/RegisterMiddleware.php(59): process
#28 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#27 vendor/flarum/core/src/Http/Middleware/SetLocale.php(51): process
#26 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#25 vendor/flarum/core/src/Http/Middleware/AuthenticateWithHeader.php(55): process
#24 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#23 vendor/flarum/core/src/Http/Middleware/AuthenticateWithSession.php(34): process
#22 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#21 vendor/flarum/core/src/Http/Middleware/RememberFromCookie.php(52): process
#20 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#19 vendor/flarum/core/src/Http/Middleware/StartSession.php(62): process
#18 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#17 vendor/flarum/core/src/Api/Middleware/FakeHttpMethods.php(31): process
#16 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#15 vendor/flarum/core/src/Http/Middleware/ParseJsonBody.php(29): process
#14 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#13 vendor/flarum/core/src/Api/Middleware/HandleErrors.php(42): process
#12 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#11 vendor/zendframework/zend-stratigility/src/MiddlewarePipe.php(83): process
#10 vendor/middlewares/request-handler/src/RequestHandler.php(84): process
#9 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#8 vendor/middlewares/base-path-router/src/BasePathRouter.php(97): process
#7 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#6 vendor/middlewares/base-path/src/BasePath.php(53): process
#5 vendor/zendframework/zend-stratigility/src/Next.php(52): handle
#4 vendor/zendframework/zend-stratigility/src/MiddlewarePipe.php(83): process
#3 vendor/zendframework/zend-stratigility/src/MiddlewarePipe.php(72): handle
#2 vendor/zendframework/zend-httphandlerrunner/src/RequestHandlerRunner.php(95): run
#1 vendor/flarum/core/src/Http/Server.php(46): listen
#0 public/index.php(22): null
```

**Comment by clarkwinkelmann** [comment]
What did you do prior to the issue ?

1. Installed
1. Enabled
1. Disabled
1. ??
1. Enabled
1. Error

Anything else happened in between ? Other extension installs ? Manually did something in the database ?

Are you able to reproduce on a clean install ?

**Comment by Hiobi** [comment]
I updated my first post with more details.
In first step, I reinstall the extension and the I tried to enable it.

I didn't try on clean insall for the moment. :)

```bash
Flarum core 0.1.0-beta.8.1
PHP version: 7.3.5-1+0~20190503093827.38+stretch~1.gbp60a41b
Loaded extensions: Core, date, libxml, openssl, pcre, zlib, filter, hash, pcntl, Reflection, SPL, sodium, session, standard, mysqlnd, PDO, xml, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, imagick, json, exif, mysqli, newrelic, pdo_mysql, pdo_sqlite, Phar, posix, readline, recode, shmop, SimpleXML, sockets, sqlite3, sysvmsg, sysvsem, sysvshm, tidy, tokenizer, wddx, xmlreader, xmlrpc, xmlwriter, xsl, zip, Zend OPcache
+---------------------------------+-----------------+------------------------------------------+
| Flarum Extensions               |                 |                                          |
+---------------------------------+-----------------+------------------------------------------+
| ID                              | Version         | Commit                                   |
+---------------------------------+-----------------+------------------------------------------+
| flarum-statistics               | v0.1.0-beta.8   |                                          |
| flarum-sticky                   | v0.1.0-beta.8   |                                          |
| flarum-subscriptions            | v0.1.0-beta.8   |                                          |
| flarum-suspend                  | v0.1.0-beta.8   |                                          |
| flarum-tags                     | v0.1.0-beta.8.2 |                                          |
| flarum-flags                    | v0.1.0-beta.8.1 |                                          |
| flarum-emoji                    | v0.1.0-beta.8   |                                          |
| flarum-bbcode                   | v0.1.0-beta.8   |                                          |
| flarum-approval                 | v0.1.0-beta.8   |                                          |
| flarum-mentions                 | v0.1.0-beta.8.1 |                                          |
| flarum-markdown                 | v0.1.0-beta.8   |                                          |
| flarum-lock                     | v0.1.0-beta.8   |                                          |
| flarum-likes                    | v0.1.0-beta.8.1 |                                          |
| flarum-pusher                   | v0.1.0-beta.8.1 |                                          |
| milescellar-lang-french         | v1.2.1          |                                          |
| noriods-auto-more               | 0.3.0           |                                          |
| flagrow-direct-links            | 0.2.0           |                                          |
| flagrow-terms                   | 0.3.1           |                                          |
| flagrow-users-list              | 0.1.2           |                                          |
| reflar-polls                    | 1.3.3           |                                          |
| flagrow-upload                  | 0.7.1           |                                          |
| flagrow-byobu                   | 0.2.2           |                                          |
| fof-socialprofile               | 0.1.1           |                                          |
| fof-pages                       | 0.2.2           |                                          |
| fof-formatting                  | 0.1.2           |                                          |
| kvothe-pipetables               | 1.0.9           |                                          |
| fof-links                       | 0.1.1           |                                          |
| kvothe-keyboard-shortcuts       | v0.1.0          |                                          |
| fof-default-group               | 0.1.0           |                                          |
| flagrow-analytics               | 0.8.0           |                                          |
| jordanjay29-summaries           | 0.3.1           |                                          |
| flagrow-user-directory          | 0.2.1           |                                          |
| amaurycarrade-syndication       | v0.2.0          |                                          |
| fof-merge-discussions           | 0.2.1           |                                          |
| flagrow-split                   | 0.3.2           |                                          |
| reflar-reactions                | 1.0.0-beta.5.2  |                                          |
| fof-prevent-necrobumping        | 0.1.3           |                                          |
| michaelbelgium-discussion-views | v0.1.0-beta.8.2 |                                          |
| fajuu-contactbutton             | v2.0            |                                          |
| fof-sentry                      | 0.1.1           |                                          |
| michaelbelgium-profile-views    | v0.1.0-beta-8-4 | d4af84a8c6ea15408e6ae00942fe4f4c9fa8288b |
| fof-recaptcha                   | 0.1.0           |                                          |
| reflar-cookie-consent           | 0.2.0           |                                          |
| fof-secure-https                | v0.1.0          |                                          |
| fof-spamblock                   | 0.2.1           |                                          |
| reflar-pwned-passwords          | 0.2.0           |                                          |
| flagrow-bazaar                  | 0.3.10          |                                          |
| fof-transliterator              | 0.1.1           |                                          |
| reflar-stopforumspam            | 0.1.5           |                                          |
| flagrow-sitemap                 | 0.2.0           |                                          |
| flarum-akismet                  | v0.1.0-beta.8   |                                          |
| flagrow-impersonate             | 0.2.0           |                                          |
| rooaaar-lang-french-extended    | dev-master      | 8059489b0a8b430961a8cf5d916a8980c5acea45 |
+---------------------------------+-----------------+------------------------------------------+
Base URL: https://forum.nefald.fr
```

**Comment by luceos** [comment]
> I reinstalled the extension

How did you re-install the extension?

**Comment by Hiobi** [comment]
Via composer :
`composer require flagrow/linguist`

**Comment by Hiobi** [comment]
Yep, it's works. I had already do that with success last days. Thanks. :)
Do you think it's relate to flagrow/linguist or just a 'normal' SQL error?

**Comment by clarkwinkelmann** [comment]
It would be interesting to find out what the error was upon uninstalling.

Because if it's what @luceos says, then the actual issue happened when uninstalling, and with the evidence so far we can't tell for sure if it's Linguist that's the culprit.

How did you uninstall the extension ? UI, Bazaar, Composer ?

**Comment by Hiobi** [comment]
It was uninstall with composer, after disabling it in flarum's admin panel.
I'm not sure but if I'm correct, it's happened after our update from beta-7 to beta-8,  three month ago, reading the documentation on [discuss.flarum.org](https://discuss.flarum.org/d/17745-flarum-0-1-0-beta-8-released).
After upgrading Flarum and updating extensions, I re-enabled them. The issue with flagrow/linguist happened at this time, I think. Or maybe it's happened before upgrading. :/

**Comment by clarkwinkelmann** [comment]
In Flarum you both disabled and uninstalled it ?

If you only disabled it, then removed it with Composer, it's very strange to have a migration error as all of the database state should have been left as-it, so enabling the extension again would just see all the migrations already done.

If you did uninstall via the Flarum admin panel before removing the extension, then likely there was an error during that. But it might not always be clear there was an error, and it's even more difficult to troubleshoot if it was in the previous Flarum version...

I feel this will be very hard to reproduce.

Didn't other people manage to get the same kind of issues with other extensions ? I have the felling I've seen a few messages like that on the forum recently.

**Comment by Hiobi** [comment]
Oh! And when I removed tables last days, only one line seems to be relate to Flagrow/Linguist in ` flarum_migrations` table. An old one > `2017_10_04_create_flagrow_linguist_strings_table`

Maybe it's related?
I didn't not remove it (I only dropped the `flarum_flagrow_linguist_strings` table) I reinstalled/re-enabled extension and it's works.

> I feel this will be very hard to reproduce.

For sure!

**Comment by clarkwinkelmann** [comment]
There should have been no need to delete any table by hand.

Though I still don't understand. If you really deleted the table but not the migration entry, Flarum shouldn't be attempting to run the migration again. Instead you'd get a table not found when Linguist tries to access its table...

**Comment by Hiobi** [comment]
Yeah me too, I’ll little bit surprise.
If I understand, it didn’t try to migrate anything after that I deleted tables.
I re-enabled extension and it works. 

Just in case, after this success I disabled and re-enabled Linguist, everything works perfectly.

**Comment by clarkwinkelmann** [comment]
I have some trouble following the steps there. Did you uninstall and reinstalled two times ? If you now say it was successful after re-enabling....

Could you summarize all the steps that lead to the unsuccessful reinstallation including the changes made to the database ? From disabling the extension to enabling it again and seeing the error ? Did you manually edit the database during that part ?

**Comment by Hiobi** [comment]
Yeah, no problem.

* **1** - At first, few months ago :
* * **1a** - I disabled all extensions
* * **1b** - I upgraded Flarum core ([following this steps](https://discuss.flarum.org/d/17745-flarum-0-1-0-beta-8-released))
* * **1c** - Updated all extensions already compatibles with beta8
* * **1d** - reinstalled Linguist too
* * **1e** - Re-enabled/enabled all extensions. Linguist cannot be enabled.
* **2** - Since 1 week, I tried to solve issue so:
* * **2a** - I uninstalled/re-installed Linguist
* * **2b** - I tried to enable it, it didn't works. I opened my issue here.
* **3** - This week-end, like I said, I dropped the table named `flarum_flagrow_linguist_strings`
* * **3a** - I didn't touch to the entry `2017_10_04_create_flagrow_linguist_strings_table` in the table `flarum_migrations`
* * **3b** - I re-enabled Linguist and it works. Problem solved.
* **4** - To be sure that to problem didn't happened again, I disabled and re-enabled Lingutis with success. No problem anymore.

**Comment by clarkwinkelmann** [comment]
I notice this issue is still open. I understand it was fixed and didn't happen again. We may never know what exactly went wrong...

I'm closing the issue :smiley:
